frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1611,6 +1613,97 @@ function ReaderHighlight:getHighlightBookmarkItem()
     end
 end
 
+function ReaderHighlight:_getInsertionPositionForRolling(page, highlight)
+    local position
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    local page_highlights = self.view.highlight.saved[page]
+    for i, cur_highlight in ipairs(page_highlights) do
+        local order = self.ui.document:compareXPointers(highlight_beginning, cur_highlight.pos0)
+        if order == 0 then
+            if highlight.highlighted and cur_highlight.highlighted then
+                order = self.ui.document:compareXPointers(highlight_end, cur_highlight.pos1)
+            end
+        end
+        if not order then
+            break
+        end
+        if order > 0 then
+            position = i
+            break
+        end
+    end
+    if position == nil then
+        position = #page_highlights + 1
+    end
+    return position
+end
+
+function ReaderHighlight:_getSortedCursorPositions(page, highlight)
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    highlight_beginning.page = page
+    highlight_end.page = page
+    local order = self.ui.document:comparePositions(highlight_beginning, highlight_end)
+    if order > 0 then
+        return highlight_beginning, highlight_end
+    else
+        return highlight_end, highlight_beginning
+    end
+end
+
+function ReaderHighlight:_getInsertionPositionForNonRolling(page, highlight)
+    local position
+    local page_highlights = self.view.highlight.saved[page]
+    local highlight_beginning, highlight_end = self:_getSortedCursorPositions(page, highlight)
+    for i, cur_highlight in ipairs(page_highlights) do
+        local cur_pos, cur_pos_end = self:_getSortedCursorPositions(page, cur_highlight)
+        local order = self.ui.document:comparePositions(highlight_beginning, cur_pos)
+        if order == 0 then
+            if highlight.highlighted and cur_highlight.highlighted then
```

Comment:
There is no such property in highlights.

Reply:
Done.